### ISSUE TITLE
Anonymous filters aggregation returns a named result when filters contain a bool query

### ISSUE THREAD

**Issue by TrustNoOne** [elastic/elasticsearch#27841]
**Elasticsearch version** (`bin/elasticsearch --version`): 6.1.0

**Plugins installed**: []

**JVM version** : 1.8.0_152

**OS version** : Mac OS X 10.13.2

**Description of the problem including expected versus actual behavior**:
Anonymous filters aggregation returns a named result using integers as keys when a bool query is specified in the filters. The result should contained an array of buckets when using anonymous filters.

**Steps to reproduce**:
This query
```json
{
  "size": 0,
  "aggs": {
    "test": {
      "filters": {
        "filters": [ { "match_all": {} } ]
      }
    }
  }
}
```

correctly returns an anonymous array of buckets:
```json
{
    "took": 10,
    "timed_out": false,
    "_shards": {
        "total": 207,
        "successful": 207,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": 1942637,
        "max_score": 0,
        "hits": []
    },
    "aggregations": {
        "test": {
            "buckets": [
                {
                    "doc_count": 1942637
                }
            ]
        }
    }
}
```

While this query:
```
{
  "size": 0,
  "aggs": {
    "test": {
      "filters": {
        "filters": [ { "bool": { } } ]
      }
    }
  }
}
```

returns a named result using array indexes as keys:

```json
{
    "took": 7,
    "timed_out": false,
    "_shards": {
        "total": 207,
        "successful": 207,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": 1942637,
        "max_score": 0,
        "hits": []
    },
    "aggregations": {
        "test": {
            "buckets": {
                "0": {
                    "doc_count": 1942637
                }
            }
        }
    }
}
``` 

In general, if a bool query is specified in the filters, I get a map having "0", "1", ... , "n" as keys instead of an array. 
The content of the bool query doesn't seem to matter, as longs as there's at least one bool query in the array of filters, the result is wrong. I used an empty bool as an example for the sake of shortness

**Comment by cbuescher** [comment]
> The content of the bool query doesn't seem to matter, as longs as there's at least one bool query in the array of filters, the result is wrong. I used an empty bool as an example for the sake of shortness

@TrustNoOne thanks for opening this issue, I can reproduce the reported behaviour for empty bool query, however if I add a clause (e.g. a "must" clause) the response is as expected (without numeric key). Just adding this as a datapoint, the behaviour for an empty bool query is still stange and needs investigation.

**Comment by TrustNoOne** [comment]
You are actually right. I thought this happened all the time, but it's only in some cases.
In my case I had a range query, this reproduces the issue as well:

```json
{
  "size": 0,
  "aggs": {
    "test": {
      "filters": {
        "filters": [
          {
            "bool": {
              "filter": {
                "range": {
                  "somefield": {
                    "gte": 170071352479318000
                  }
                }
              }
            }
          }
        ]
      }
    }
  }
}
```

**Comment by cbuescher** [comment]
So this is what I think happens here: the empty bool query gets rewritten internally to a MatchAllQueryBuilder, and as a follow up FiltersAggregationBuilder#doRewrite creates a new FiltersAggregationBuilder which doesn't correctly copy the original "keyed" field. I think that's the root cause of this and I will start working on a fix.

**Comment by cbuescher** [comment]
@TrustNoOne thanks for reporting this, good catch. And thanks for the detailed description, it was easy to reproduce that way. Fix will be in next 6.1 release and onwards.

**Comment by TrustNoOne** [comment]
sure thing! thanks for fixing it that fast
